### PR TITLE
Add basic config versioning

### PR DESCRIPTION
**Pull request by conradoverta**

### CODE REVIEW

File: protos/protos/public/modeldb/versioning/Config.proto [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    repeated HyperparameterSetConfigBlob hyperparameter_set = 1;
+    repeated HyperparameterConfigBlob hyperparameters = 2;
+}
+
+message HyperparameterConfigBlob {
+    string name = 1;
+    HyperparameterValuesConfigBlob value = 2;
+}
+
+message HyperparameterValuesConfigBlob {
```
Comment (conradoverta) [inline protos/protos/public/modeldb/versioning/Config.proto]: Fold into the previous one. No need for a separate message here?

**Comment by ravishetye** [comment]
based on the previous two PRs this needs to be a draft as well?

File: protos/protos/public/modeldb/versioning/Config.proto [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        float float_value = 2;
+        string string_value = 3;
+    }
+}
+
+message HyperparameterSetConfigBlob {
+    string name = 1;
+    HyperparameterValuesConfigBlob interval_begin = 2;
+    HyperparameterValuesConfigBlob interval_end = 3;
+    repeated HyperparameterValuesConfigBlob discrete = 4;
```
Comment (ravishetye) [inline protos/protos/public/modeldb/versioning/Config.proto]: may be extract the begin and end to a separate message `HyperparameterValuesContinousConfigBoundaries` and then a one of between continuous and discrete?
not sure but do you think begin, end and step size support will be useful?

Comment (conradoverta) [reply protos/protos/public/modeldb/versioning/Config.proto]: Makes sense. There could be a sampling strategy kind of thing.

**Pull request merged**